src/util/win32_utils.cc

```diff
@@ -0,0 +1,18 @@
+#include "win32_utils.h"
+
+#define NOMINMAX
+#include <windows.h>
+
+namespace jumanpp {
+namespace util {
+
+std::wstring to_wide_string(const StringPiece str) {
+  int str_size = static_cast<int>(str.size());
+  int size = MultiByteToWideChar(CP_UTF8, 0, str.data(), str_size, NULL, 0);
+  std::wstring result(size, 0);
+  MultiByteToWideChar(CP_UTF8, 0, str.data(), str_size, &result[0], size);
```

Comment:
`&result[0]` should be also result.data()

I wonder how this plays with the null-terminator. c_str() must return null-terminated strings by C++11, but we just copy the stuff here inside the string.
Probably we are safe here and the std::string will do the right thing.

Maybe should write a test for this some day.

Reply:
Not sure i understand your concerns... MultiByteToWideChar returns size that includes null-terminating character.
So `wstring(size)` here will be written as `<wtf-16>0`

Reply:
Please note that `result.data` returns const pointer which cannot be used unless you perform const_cast which is not entirely correct way to use this method.

Reply:
Additional note: `.data()` and `.c_str()` are the same starting from C++11 i.e. they return null-terminated array

Reply:
I made some inquiry: it seems `string.c_str()` appends 0, if there is none

Reply:
I made some more inquiries. It seem `std::string` always append 0 at the end that is not considered in `std::string.size()`.

Which means when we `std::string(size, 0)` we actually have string with size = size + 1(hidden 0)
So even if passed string's slice has no 0, there will be still 0 in `.c_str()` otherwise we'll have just two 0 at the end, which is not critical I guess?

Reply:
Ah, ok, non-const data() is C++17 as it seems.
http://en.cppreference.com/w/cpp/string/basic_string/data
Nevermind my first comment then.

Yep, because C++11 requires `c_str()` to return null-terminated string, the actual string buffer should be at least 1 character larger than the specified (I think it should be actually observable via `capacity()`).

NVM my comment, LGTM.